aws/resource_aws_spot_fleet_request.go

```diff
@@ -1133,32 +1139,6 @@ func rootBlockDeviceToSet(
 	return set
 }
 
-func resourceAwsSpotFleetRequestUpdate(d *schema.ResourceData, meta interface{}) error {
-	// http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifySpotFleetRequest.html
-	conn := meta.(*AWSClient).ec2conn
-
-	d.Partial(true)
-
-	req := &ec2.ModifySpotFleetRequestInput{
-		SpotFleetRequestId: aws.String(d.Id()),
-	}
-
-	if val, ok := d.GetOk("target_capacity"); ok {
-		req.TargetCapacity = aws.Int64(int64(val.(int)))
-	}
-
-	if val, ok := d.GetOk("excess_capacity_termination_policy"); ok {
-		req.ExcessCapacityTerminationPolicy = aws.String(val.(string))
-	}
-
-	resp, err := conn.ModifySpotFleetRequest(req)
-	if err == nil && aws.BoolValue(resp.Return) {
```

Comment:
I think this was the original cause for the staticcheck error, where it needed to be updated to something like:

```go
if err != nil {
  return fmt.Errorf("error modifying EC2 Spot Fleet Request (%s): %s", d.Id(), err)
}
```

Reply:
Updated to address original error.